mjcf_to_sdformat/mjcf_to_sdformat/converters/geometry.py

```diff
@@ -0,0 +1,130 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module to convert MJCF geoms to SDFormat Collision/Visual geometries"""
+
+from ignition.math import Vector2d, Vector3d
+
+import sdformat as sdf
+import sdformat_mjcf_utils.sdf_utils as su
+
+VISUAL_NUMBER = 0
+COLLISION_NUMBER = 0
+
+
+def add_mjcf_geometry_to_sdf(geom):
+    """
+    Converts an MJCF geom to a SDFormat geometry.
+
+    :param mjcf.Element geom: The MJCF geom
+    :return: The newly created SDFormat geometry.
+    :rtype: sdf.Geometry
+    """
+    sdf_geometry = sdf.Geometry()
+    if geom.type == "box":
+        box = sdf.Box()
+        box.set_size(su.list_to_vec3d(geom.size) * 2)
+        sdf_geometry.set_box_shape(box)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.BOX)
+    elif geom.type == "capsule":
+        capsule = sdf.Capsule()
+        capsule.set_radius(geom.size[0])
+        if geom.fromto is None:
+            capsule.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            capsule.set_length(length)
+        sdf_geometry.set_capsule_shape(capsule)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CAPSULE)
+    elif geom.type == "cylinder":
+        cylinder = sdf.Cylinder()
+        cylinder.set_radius(geom.size[0])
+        if geom.fromto is None:
+            cylinder.set_length(geom.size[1] * 2)
+        else:
+            v1 = Vector3d(geom.fromto[0], geom.fromto[1], geom.fromto[2])
+            v2 = Vector3d(geom.fromto[3], geom.fromto[4], geom.fromto[5])
+            length = v1.distance(v2)
+            cylinder.set_length(length)
+        sdf_geometry.set_cylinder_shape(cylinder)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.CYLINDER)
+    elif geom.type == "ellipsoid":
+        ellipsoid = sdf.Ellipsoid()
+        ellipsoid.set_radii(su.list_to_vec3d(geom.size))
+        sdf_geometry.set_ellipsoid_shape(ellipsoid)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.ELLIPSOID)
+    elif geom.type == "sphere":
+        sphere = sdf.Sphere()
+        sphere.set_radius(geom.size[0])
+        sdf_geometry.set_sphere_shape(sphere)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.SPHERE)
+    elif geom.type == "plane":
+        plane = sdf.Plane()
+        plane.set_size(Vector2d(geom.size[0] * 2, geom.size[1] * 2))
+        sdf_geometry.set_plane_shape(plane)
+        sdf_geometry.set_type(sdf.Geometry.GeometryType.PLANE)
+    else:
+        raise RuntimeError(
+            f"Encountered unsupported shape type {geom.type}")
+    return sdf_geometry
+
+
+def add_mjcf_visual_to_sdf(geom):
```

Comment:
This seems to be following the naming pattern of the functions in `sdformat_to_mjcf.convert`. However, those functions take a body and **add** something to the body. These functions create new objects and return them, so I don't think we should call them `add_*`. Alternatively, we can change these function to take an object. For example, in this case, it would be `add_mjcf_visual_to_sdf(visual, geom)`.

Reply:
https://github.com/gazebosim/gz-mujoco/pull/26/commits/21b76c3da90666223352811af70d216f8f722e68